pkg/client/unversioned/request.go

```diff
@@ -642,6 +642,7 @@ func (r *Request) Watch() (watch.Interface, error) {
 		return nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
+		defer resp.Body.Close()
```

Comment:
It seems `resp.Body` is not used out side of transformResponse. Could you move it to where it's used inside transformResponse? Thanks.